### DISCUSSION TITLE
Can use C++ language in Lua editor？

### DISCUSSION THREAD

**Discussion by southeast2017** [o3de/o3de#7941] [Help Wanted] [answered]
In a Lua Script，Can I use C++ and lua to write a script？？

**Comment by BytesOfPiDev** [comment, marked as answer]
You cannot write C++ code within Lua.

You cannot script with C++.

You can, however, write C++ code and expose it to Lua using the AZ::BehaviorContext: https://www.o3de.org/docs/user-guide/programming/components/reflection/behavior-context/
